clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -321,20 +356,36 @@ private String getTopicName() {
         return new HonoTopic(HonoTopic.Type.COMMAND_INTERNAL, adapterInstanceId).toString();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Closes the Kafka admin client and consumer.
+     *
+     * @return A future indicating the outcome of the operation.
+     *         The future will be succeeded once this component is stopped.
+     */
     @Override
     public Future<Void> stop() {
-        retryCreateTopic.set(false);
-        vertx.cancelTimer(retryCreateTopicTimerId);
 
-        if (consumer == null) {
-            return Future.failedFuture("not started");
+        if (lifecycleStatus.isStopped()) {
+            // nothing to do
+            return Future.succeededFuture();
         }
 
-        return CompositeFuture.all(closeAdminClient(), closeConsumer())
-                .mapEmpty();
+        final var result = lifecycleStatus.newStopAttempt();
+        retryCreateTopic.set(false);
+        vertx.cancelTimer(retryCreateTopicTimerId);
+
+        CompositeFuture.all(closeAdminClient(), closeConsumer())
+            .map((Void) null)
+            .onSuccess(ok -> lifecycleStatus.setStopped());
+        return result.future();
```

Comment:
In contrast to the previous solution, the code above doesn't skip running the actual stopping logic if `lifecycleStatus.isStopping()` is `true` already.

How about a solution where the stopping logic is passed as a parameter?
```
    @Override
    public Future<Void> stop() {
        return lifecycleStatus.newStopAttempt(() -> {
            retryCreateTopic.set(false);
            vertx.cancelTimer(retryCreateTopicTimerId);
            return CompositeFuture.all(closeAdminClient(), closeConsumer())
                    .mapEmpty();
        });
    }
```

With the new LifecycleStatus method looking like this:
```
    public synchronized Future<Void> newStopAttempt(final Supplier<Future<Void>> stopAction) {

        if (isStopped()) {
            // nothing to do
            return Future.succeededFuture();
        }

        final Promise<Void> result = Promise.promise();
        addOnStoppedHandler(result);
        if (!isStopping()) {
            setStopping();
            stopAction.get()
                    .onComplete(done -> setStopped());
        }
        return result.future();
    }
```

The part where `setStopped()` gets invoked:
```
            stopAction.get()
                    .onComplete(done -> setStopped());
```
should be put in an `onComplete` instead of `onSuccess` FMPOV. 
Otherwise a failure running the `stopAction` above will let the `stop()` method result future never be completed.
WDYT?

Reply:
> In contrast to the previous solution, the code above doesn't skip running the actual stopping logic if lifecycleStatus.isStopping() is true already.

shoot, that's true

Reply:
> Otherwise a failure running the stopAction above will let the stop() method result future never be completed.

Yes, we should reach the stopped state in any case.